### PR TITLE
#69 Force create user config for email otp

### PR DESCRIPTION
**Pull request by rtufisi**

### CODE REVIEW

File: src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java [hunk 1]
```diff
@@ -38,15 +45,26 @@ private void challenge(AuthenticationFlowContext context, FormMessage errorMessa
...
+    EventBuilder event = context.newEvent();
+    UserModel user =
+            MagicLink.getOrCreate(
+                    context.getSession(),
+                    context.getRealm(),
+                    email,
+                    isForceCreate(context, false),
+                    false,
+                    false,
+                    MagicLink.registerEvent(event, "email-otp"));
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java:65]: Can "email-otp" be a static?

File: src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java [hunk 2]
```diff
@@ -94,4 +112,26 @@ public void setRequiredActions(KeycloakSession session, RealmModel realm, UserMo
 
   @Override
   public void close() {}
+
+  public static String getAttemptedUsername(AuthenticationFlowContext context) {
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java:116]: Seems like we can make this a utility method that both classes can use

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java [hunk 3]
```diff
@@ -79,7 +79,7 @@ public void action(AuthenticationFlowContext context) {
             isForceCreate(context, false),
             isUpdateProfile(context, false),
             isUpdatePassword(context, false),
-            MagicLink.registerEvent(event));
+            MagicLink.registerEvent(event, "magic"));
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkAuthenticator.java:82]: make this a static

File: src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java [hunk 4]
```diff
@@ -121,7 +121,7 @@ public void action(AuthenticationFlowContext context) {
             isForceCreate(context, false),
             false,
             false,
-            MagicLink.registerEvent(event));
+            MagicLink.registerEvent(event, "magic"));
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/auth/MagicLinkContinuationAuthenticator.java:124]: same. make this a static

File: src/main/java/io/phasetwo/keycloak/magic/resources/MagicLinkResource.java [hunk 5]
```diff
@@ -58,7 +58,7 @@ public MagicLinkResponse createMagicLink(final MagicLinkRequest rep) {
             forceCreate,
             updateProfile,
             updatePassword,
-            MagicLink.registerEvent(event));
+            MagicLink.registerEvent(event, "magic"));
```
Comment (xgp) [inline src/main/java/io/phasetwo/keycloak/magic/resources/MagicLinkResource.java:61]: same. make this a static

**Review by xgp — CHANGES_REQUESTED** [review]
looks good. a few small changes.

File: src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java (hunk 1 above)
Comment (rtufisi) [reply src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java]: Done

File: src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java (hunk 2 above)
Comment (rtufisi) [reply src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java]: Done

**Review by xgp — APPROVED** [review]

**Pull request merged**